src/newsletter-editor/sidebar/send-to.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * WordPress dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { FormTokenField, Button, ButtonGroup, Notice } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { Icon, external } from '@wordpress/icons';
+
+const SendTo = ( { availableItems, formLabel, onChange, placeholder, reset, selectedItem } ) => {
+	const [ isEditing, setIsEditing ] = useState( false );
+	const [ isUpdating, setIsUpdating ] = useState( false );
+	const [ error, setError ] = useState( false );
+
+	return (
+		<div className="newspack-newsletters__send-to">
+			{ error && ! isEditing && (
+				<Notice status="error" isDismissible={ false }>
+					{ error }
+				</Notice>
+			) }
+
+			{ selectedItem && ! isEditing ? (
+				<>
+					<p className="newspack-newsletters__send-to-details">
+						{ selectedItem.name }
+						<span>
+							{ isUpdating
+								? sprintf(
+										// Translators: Shown while resetting the selected send-to item. %s is the item type.
+										__( 'Resetting %s…', 'newspack-newsletters' ),
+										selectedItem.typeLabel.toLowerCase()
+								  )
+								: selectedItem.typeLabel }
+							{ ! isUpdating && selectedItem.hasOwnProperty( 'count' )
+								? ' • ' +
+								  sprintf(
+										// Translators: If available, show a contact count alongside the selected item's type. %d is the number of contacts in the item.
+										_n( '%d contact', '%d contacts', selectedItem.count, 'newspack-newsletters' ),
+										selectedItem.count.toLocaleString()
+								  )
+								: '' }
+						</span>
+					</p>
+				</>
+			) : (
+				<>
+					<FormTokenField
+						disabled={ isUpdating }
+						label={ formLabel || __( 'Select a list', 'newspack' ) }
+						maxSuggestions={ 10 }
+						onChange={ items => {
+							setError( false );
+							setIsUpdating( true );
+							onChange( items )
+								.catch( e => {
```

Comment:
If I select an item, click to edit, and select the same item again I get the following error and the input fails to change:

```
Cannot read properties of undefined (reading 'catch')
```

Reply:
I think we'll fix this in the next round of changes in #1605